cosmos/dbt/graph.py

```diff
@@ -605,6 +642,11 @@ def load_from_dbt_manifest(self) -> None:
                     file_path=self.execution_config.project_path / Path(node_dict["original_file_path"]),
                     tags=node_dict["tags"],
                     config=node_dict["config"],
+                    has_freshness=(
+                        node_dict["freshness"] is not None
+                        if node_dict["resource_type"] == "source" and "freshness" in node_dict
+                        else False
+                    ),
```

Comment:
```suggestion
                    has_freshness= node_dict["resource_type"] == "source" and node_dict.get("freshness")
```

Reply:
Ended up adding it like this
```
has_freshness=(
                        is_freshness_effective(node_dict.get("freshness"))
                        if DbtResourceType(node_dict["resource_type"]) == DbtResourceType.SOURCE
                        else False
                    ),
```

since both manifest and `ls` can return an empty freshness dictionary
```
{
                "warn_after": {
                    "count": null,
                    "period": null
                },
                "error_after": {
                    "count": null,
                    "period": null
                },
                "filter": null
            }
```